src/uart/mod.rs

```diff
@@ -17,17 +19,22 @@ pub struct UART2 {}
 
 impl private::UARTPortCompatable for UART0 {
     const PORT_PTR: usize = mmio::UART_0;
+    const PORT_NUM: usize = 0;
 }
 impl private::UARTPortCompatable for UART1 {
     const PORT_PTR: usize = mmio::UART_1;
+    const PORT_NUM: usize = 1;
 }
 impl private::UARTPortCompatable for UART2 {
     const PORT_PTR: usize = mmio::UART_2;
+    const PORT_NUM: usize = 2;
 }
 
+#[allow(dead_code)]
 pub struct UART<Port = NoPort> {
     reg: registers::Registers,
     ph: PhantomData<Port>,
+    gpio: [GpioPin; 2],
 }
 
 #[allow(unused)]
```

Comment:
Since these are public, it should not complain about `#[allow(unused)]` so this can be removed.